src/main/java/com/compilercharisma/chameleonbusinessstudio/service/AppointmentService.java

```diff
@@ -114,14 +114,14 @@ public boolean isAppointmentValid(AppointmentEntity e){
         return validator.isValid(e);
     }
     
-    private void registerUser(AppointmentEntity appt, String email){
+    public void registerUser(AppointmentEntity appt, String email){
         if(!isSlotAvailable(appt)){
             throw new UnsupportedOperationException(String.format("Cannot register \"%s\" in appointment#%d: no slots available", email, appt.getId()));
```

Comment:
Nice exception handling, would it be possible to do so for when we expect bad requests?

Reply:
Define "do so"? Do you mean provide error messages like line 119?

Reply:
Yeah, that's what I meant